docs/rfc/rfc-1.adoc

```diff
@@ -435,13 +435,64 @@ As time passes and operators drop out of the system, a wallet becomes at risk
 of being able to meet the 51-of-100 threshold to produce signatures.
 Additionally, we want to avoid situations where operators are the custodians of
 a wallet for extended periods. To avoid these issues, once a wallet is older
-than the `wallet_max_age`, or if it drops below the liveness threshold (say, below 70 on
-a <<heartbeat,heartbeat>>), we motion to transfer the funds to another randomly
-selected wallet.
+than the `wallet_max_age`, or if it drops below the liveness threshold (say,
+below 70 on a <<heartbeat,heartbeat>>), we motion to
+<<closing-a-wallet,transfer the funds>> to another randomly selected wallet.
 
 Once a wallet no longer has funds and is not the primary wallet for new
-deposits, it can be closed and operators are no longer required to maintain
-it.
+deposits, it can be <<closing-a-wallet,closed>> and operators are no longer
+required to maintain it.
+
+[[closing-a-wallet]]
+==== Closing A Wallet
+
+Governable Parameters:
+
+- `heartbeat`: The number of group members required for a
+  <<heartbeat,heartbeat>> to be successful.
+- `random_transfer_depth`: The depth (sorted youngest to oldest) at which we
+  randomly select a new wallet to be the custodian of the old wallet's funds
+  when closing a wallet. For instance, a depth of 3 means that the 3 youngest
+  wallets can be randomly chosen.
+- `wallet_closure_timeout`: The amount of time that a wallet has to
+  successfully move its funds to another wallet and inform the ethereum chain
+  before it is at risk of punishment.
+- `wallet_max_age`: The oldest we allow a wallet to become before we transfer the funds
+  to a randomly selected wallet.
+- `wallet_min_btc`: The smallest amount of btc a wallet can hold after a
+  redemption before we attempt to close the wallet and transfer the funds to a
+  randomly selected wallet.
+
+When a wallet drops below the `heartbeat` threshold during a
+<<heartbeat,heartbeat>> or ends a <<redemption,redemption>> with less than
+`wallet_min_btc` remaining, or exceeds the `wallet_max_age`, the operators
+begin the process of closing it. If there are still funds remaining, the
+<<operator-only,first operator>> posts a <<operator-only,reimbursable>>
+transaction to the ethereum chain declaring an intention to close the wallet.
+
+That operator proposes a BTC fee (as in <<sweeping,sweeping>>) and the
+operators use the hash of the previous intention-to-close transaction as a
+source of randomness to choose viable (it must also have a passing heartbeat
+and be above `wallet_min_btc`, and be within the `random_transfer_depth`)
+wallet to transfer the remaining funds to. They construct a P2PKH transaction
+moving the wallet's main UTXO to that wallet.
+
+After this transaction is complete, <<public-knowledge,the public>> is able to
+submit a SPV proof to let ethereum know that the funds were transferred
+nonfraudulently, and which account balances need to be updated (like the
+treasury which pays the bitcoin mining fee).
```

Comment:
I think we should do the bookkeeping of Bitcoin fees off-chain to minimise gas costs. I think it is enough to emit an event with Bitcoin transaction hash so that the treasury governance can do the donation (see https://github.com/keep-network/tbtc-v2/pull/132/files#diff-1044c578112615d9cb745f17e8203a444e72ddb3c5d8d5fd97d97038952c77b1R622) after some time.

Reply:
addressed in 73c7f373212d240c743d6664dbcc01342116dbd9